### PR TITLE
v3.4.8

### PR DESCRIPTION
**Pull request by ClaraTersi**
# Midaz Pull Request Checklist

## Pull Request Type
[//]: # (Check the appropriate box for the type of pull request.)

- [ ] Console
- [ ] Documentation
- [ ] Infra
- [ ] Mdz
- [x] Onboarding
- [ ] Pipeline
- [x] Transaction
- [ ] Tests

## Checklist
Please check each item after it's completed.

- [x] I have tested these changes locally.
- [x] I have updated the documentation accordingly.
- [x] I have added necessary comments to the code, especially in complex areas.
- [x] I have ensured that my changes adhere to the project's coding standards.
- [x] I have checked for any potential security issues.
- [x] I have ensured that all tests pass.
- [x] I have updated the version appropriately (if applicable).
- [x] I have confirmed this code is ready for review.

## Additional Notes
[//]: # (Add any additional notes, context, or explanation that could be helpful for reviewers.)

## Summary by CodeRabbit

* **New Features**
  * Idempotency cache for faster duplicate-transaction detection and reduced repeated work.
  * Distributed consumer locking to reduce concurrent processing conflicts.
  * Service health checks for gRPC dependencies with configurable timeout, surfacing service-unavailable early.

* **Bug Fixes**
  * Safer transaction queue/backup error handling with explicit failure codes and clearer retry behavior.
  * Idempotent-friendly handling of duplicate inserts and cache-backed reads.
  * Defaulting missing balance keys to avoid edge-case failures.

* **Improvements**
  * Account-alias sanitization for sent transactions.
  * Better MongoDB/proto address defaults and updated example app/frontend versions.

* **Tests**
  * Unit tests covering idempotency mapping and cache behaviors.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: components/transaction/internal/bootstrap/redis.consumer.go [hunk 1]
```diff
@@ -129,12 +136,40 @@ Outer:
...
+				logger.Infof("Message %s already being processed by another pod, skipping", key)
+
+				return
+			}
+
+			if m.Validate == nil {
+				logger.Warnf("Message (key: %s) has nil Validate field, skipping. Message will remain in queue.", key)
+
+				return
+			}
```
[the remark this answers is not in the document]
Comment (ClaraTersi) [reply components/transaction/internal/bootstrap/redis.consumer.go:164]: Expected behavior for now. Improvements will follow in future versions.

**Review by Ygohr — APPROVED** [review]
